plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -1079,6 +1042,65 @@ public void deployModel(
         }
     }
 
+    private void deployRemoteOrBuiltInModel(MLModel mlModel, Integer eligibleNodeCount, ActionListener<String> wrappedListener) {
+        String modelId = mlModel.getModelId();
+        setupModelRateLimiter(modelId, eligibleNodeCount, mlModel.getModelRateLimiterConfig());
+        if (mlModel.getConnector() != null || FunctionName.REMOTE != mlModel.getAlgorithm()) {
+            setupParamsAndPredictable(modelId, mlModel);
+            mlStats.getStat(MLNodeLevelStat.ML_DEPLOYED_MODEL_COUNT).increment();
+            modelCacheHelper.setModelState(modelId, MLModelState.DEPLOYED);
+            wrappedListener.onResponse("successful");
+            return;
+        }
+        log.info("Set connector {} for the model: {}", mlModel.getConnectorId(), modelId);
+        getConnector(mlModel.getConnectorId(), ActionListener.wrap(connector -> {
+            mlModel.setConnector(connector);
+            setupParamsAndPredictable(modelId, mlModel);
+            mlStats.getStat(MLNodeLevelStat.ML_DEPLOYED_MODEL_COUNT).increment();
+            modelCacheHelper.setModelState(modelId, MLModelState.DEPLOYED);
+            wrappedListener.onResponse("successful");
+            log.info("Completed setting connector {} in the model {}", mlModel.getConnectorId(), modelId);
+        }, wrappedListener::onFailure));
+    }
+
+    private void setupParamsAndPredictable(String modelId, MLModel mlModel) {
+        Map<String, Object> params = setUpParameterMap(modelId);
+        Predictable predictable = mlEngine.deploy(mlModel, params);
+        modelCacheHelper.setPredictor(modelId, predictable);
+    }
+
+    private Map<String, Object> setUpParameterMap(String modelId) {
+        TokenBucket modelRateLimiter = getModelRateLimiter(modelId);
+        Map<String, TokenBucket> userRateLimiterMap = getUserRateLimiterMap(modelId);
+
+        Map<String, Object> params = new HashMap<>() {
```

Comment:
Let's use the standard way to initialize HashMap 
```
Map<String, Object> params = new HashMap<>();
params.put(ML_ENGINE, mlEngine);
params.put(SCRIPT_SERVICE, scriptService);
params.put(CLIENT, client);
```

Reply:
Ok